core/src/main/scala/zio/jdbc/Query.scala

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2022 John A. De Goes and the ZIO Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package zio.jdbc
+
+import zio._
+import zio.stream._
+
+final class Query[+A](val sql: SqlFragment, val decode: ZResultSet => A) {
```

Comment:
Is there any reason to avoid making this a `final case class`?

Reply:
I don't think there is, I just was following the definition of the original `Sql` class. From my perspective, it is even better to make it a case class.